Applications/DataHolderLib/Project.cpp

```diff
@@ -99,4 +99,53 @@ bool Project::getUniqueName(std::string &name) const
     return isUnique;
 }
 
+void Project::removePrimaryVariable(std::string const primary_var_name)
+{
+    std::size_t n_bc(_boundary_conditions.size());
+    for (std::size_t i = 0; i < n_bc; ++i)
+        if (_boundary_conditions[i]->getProcessVarName() == primary_var_name)
```

Comment:
since `i` is used only to access the boundary condition, a for-range-loop is more appropriate.